wordpress-mobile/WordPress-Android
# [Home Page Picker] Move to Page sheet

**Fixes** https://github.com/wordpress-mobile/WordPress-iOS/pull/15293#issuecomment-730520774

Depends on: https://github.com/wordpress-mobile/WordPress-Android/pull/13424

## Description
This PR applies a dismissible modal behavior to the site creation flow

## To test:

<details>
<summary><strong>To disable or enable the development version of Home Page Picker</strong></summary>

- Open the app from the build that allows FeatureFlags such as a PR build or a local development build
    - By default, the modal layout picker is enabled only in development builds.
    - From the site page:
        1. Click on your Gravatar.
        1. Click on App Settings.
        1. Click on Test feature configuration.
        1. Toggle "HomePagePickerFeatureConfig" to enable or disable the picker
        1. Restart the App
</details>

<details>
<summary><strong>To start the Home Page Picker screen</strong></summary>

1. Start the site creation flow (e.g. *Choose site > Add button*)
1. Select the *Create WordPress.com site* option
</details>

1. Start the site creation flow
1. In each step verify that the flow is dismissible by swiping down from the top 

## Screenshots

|Normal Mode|Dark Mode|
|---|---|
|![device-2020-11-20-210434](https://user-images.githubusercontent.com/304044/99840101-93c08300-2b74-11eb-8d78-28a41bbde051.png)|![device-2020-11-20-210454](https://user-images.githubusercontent.com/304044/99840069-83a8a380-2b74-11eb-975b-0688559c687a.png)|

PR submission checklist:
